### ISSUE TITLE
merge bot: special case bump version patch for new modules

### ISSUE THREAD

**Issue by sbidoul** [OCA/oca-github-bot#54]
When a new module is merged, the bot currently does not bump version.
In case the commit that set the version number is not the last commit, then a .dev version goes to pypi.

To avoid this, the bot could detect that situation and force a patch version bump if the commit setting the version number is not the last one.

**Comment by sbidoul** [comment]
cc/ @rousseldenis

**Comment by pedrobaeza** [comment]
I see possible conflicts if the author of the PR already does the bump, isn't it?

**Comment by sbidoul** [comment]
That is only for new modules (and migrations), so that should not be a problem?

**Comment by pedrobaeza** [comment]
So the author puts `12.0.1.0.0`, the bot is going to put `12.0.1.0.1`?

**Comment by sbidoul** [comment]
Yes, but only if the author (or the bot) made additional commits after the commit that sets 12.0.1.0.0.

**Comment by pedrobaeza** [comment]
Well, you know I'm a bit frustrated by all of these weird requirements for pip deployment that are out of Odoo itself, requiring a lot of effort (no matter if it's yours or from other contributors), but let's not have again this discussion. Go ahead if it's the only solution.

**Comment by sbidoul** [comment]
Well it's not only for pip. it's a way to give a better meaning to version numbers. Could be useful for the OCA apps store for instance.

**Comment by pedrobaeza** [comment]
I don't see this specific case for that, as `12.0.1.0.0` is perfectly good for the rest. I suppose this is for avoiding the `--pre` problem in pypi.

**Comment by sbidoul** [comment]
The real question is: which commit is 12.0.1.0.0 exactly?

**Comment by pedrobaeza** [comment]
Not always the last, but that doesn't mean `12.0.1.0.0` is not valid. Several valid cases:

* The migration is done some time ago, and a reviewer ask to include a later pull request made on a previous version that applies also here. The commit is cherry-pick, but the migration is still the same.
* Migrator adds a new feature while migrating the module and decides to put it in a separate commit. `12.0.1.0.0` is still valid as there has not been previous version without that feature.

In that cases, `12.0.1.0.1` sounds ugly, as there hasn't been ever `12.0.1.0.0` version, but as said, if it's the only solution, go ahead.

**Comment by emagdalenaC2i** [comment]
What happens with this type of migrations? https://github.com/OCA/server-tools/pull/1587/files Here the version is set to the same number than the fontawesome used?

**Comment by sbidoul** [comment]
@emagdalenaC2i interesting point

**Comment by rousseldenis** [comment]
@sbidoul What's the status of this ?

It's sometimes frustrating, when explaining to new mergers behaviour that pushes dev versions to pypi when using nobump instead of patch and attracting people that don't like no round version number for new modules (or migrated ones). See https://github.com/OCA/crm/pull/365#issuecomment-744604513
